File: Python_Codes/6. Collections Framework/Collections/QueueExample.py
from collections import deque

def queue_operations():
    # Create a queue using deque
    queue = deque()

    # Add elements to the queue
    queue.append("First")
    queue.append("Second")
    queue.append("Third")

    # Display the queue
    print("Queue:", list(queue))

    # Remove an element from the queue
    removed_element = queue.popleft()
    print("Removed Element:", removed_element)

    # Display the queue after removal
    print("Queue after removal:", list(queue))

    # Peek at the front of the queue
    front_element = queue[0] if queue else None
    print("Front Element:", front_element)

    # Check if the queue is empty
    is_empty = len(queue) == 0
    print("Is the queue empty?", is_empty)


if __name__ == "__main__":
    queue_operations()

File: Projects/JavaProject/src/com/Collections_Framework/QueueExample.java
package com.Collections_Framework;

import java.util.LinkedList;
import java.util.Queue;

public class QueueExample {

    // Function to demonstrate queue operations
    public static void queueOperations() {
        // Create a queue using LinkedList
        Queue<String> queue = new LinkedList<>();

        // Add elements to the queue
        queue.add("First");
        queue.add("Second");
        queue.add("Third");

        // Display the queue
        System.out.println("Queue: " + queue);

        // Remove an element from the queue
        String removedElement = queue.poll();
        System.out.println("Removed Element: " + removedElement);

        // Display the queue after removal
        System.out.println("Queue after removal: " + queue);

        // Peek at the front of the queue
        String frontElement = queue.peek();
        System.out.println("Front Element: " + frontElement);

        // Check if the queue is empty
        boolean isEmpty = queue.isEmpty();
        System.out.println("Is the queue empty? " + isEmpty);
    }

    public static void main(String[] args) {
        // Call the function to demonstrate queue operations
        queueOperations();
    }
}

